addon/utils/locale.js

```diff
@@ -87,10 +87,11 @@ export default class Locale {
   }
 
   _defineMissingTranslationTemplate(key) {
+    const I18n = this.container.lookupFactory('service:i18n');
     const missingMessage = this.container.lookupFactory('util:i18n/missing-message');
     const locale = this.id;
 
-    function missingTranslation(data) { return missingMessage(locale, key, data); }
+    function missingTranslation(data) { return missingMessage.call(I18n, locale, key, data); }
```

Comment:
The idea was to consolidate the public API. Currently, the way to override the functionality is to define a `utils/i18n/missing-message.js` in your app. If it becomes a public method of `service:i18n`, then it, like every other override, happens in `app/services/i18n.js`. But I don't think it's a strong requirement. We can always move it later.